vwifi.c

```diff
@@ -599,7 +600,9 @@ static void owl_disconnect_routine(struct work_struct *w)
             return;
         }
 
-        list_del(&vif->bss_list);
+        if (vif->ap->ap_enable && !list_empty(&vif->bss_list))
```

Comment:
If AP call `stop_ap()`, the connected STA will be kicked out of the `bss_list` by `list_del()`. But if AP call  `start_ap()` again, then STA tries to call `disconnect()`, and unfortunately the expression `vif->ap->ab_enable && !list_empty(&vif->bss_list)` will be true since `vif->ap->ap_enable` is true and `!list_empty(&vif->bss_list)` is true. (so `list_empty(&vif->bss_list)` is false, because `&(vis->bss_list)->next` is not `&vis->bss_list` (see `list_del()` in `includ/linux/list.h`, the deleted `list_head`'s `next` and `prev` will be invalid memory area rather than the `list_head` itself)). Then `list_del()` will be called again (in `stop_ap()` is first time), and kernel could panic since we dereference the invalid memory area. 

Maybe we could use `list_del_init_careful()` in `stop_ap()`, and the problem should be solved.